tools/c7n_azure/c7n_azure/resources/key_vault.py

```diff
@@ -114,3 +119,48 @@ def enhance_policies(self, access_policies):
             policy['principalName'] = GraphHelper.get_principal_name(aad_object)
 
         return access_policies
+
+
+@resources.register('key')
+class KeyVaultKey(ArmResourceManager):
+
+    class resource_type(ArmResourceManager.resource_type):
+        service = 'azure.mgmt.keyvault'
+        client = 'KeyVaultManagementClient'
+        enum_spec = ('vaults', 'list', None)
+
+    def augment(self, resources):
+        url_base = 'vault.azure.net'
+        s = Session(resource='https://' + url_base)
+        self.keyvault_client = KeyVaultClient(s.get_credentials())
+        r = []
+        for index, item in enumerate(resources):
+            vault_url = 'https://' + resources[index]["name"] + '.vault.azure.net'
+            try:
+                vkeys = self.keyvault_client.get_keys(vault_url)
+                for vkey in vkeys:
+                    r.append({'kid': vkey.kid,
+                              'keyvault': {'id': resources[index]["id"],
+                                           'name': resources[index]["name"],
+                                           'location': resources[index]["location"],
+                                           'tags': resources[index]["tags"]},
+                              'enabled': vkey.attributes.enabled,
+                              'created': vkey.attributes.created,
+                              'updated': vkey.attributes.updated,
+                              'tags': vkey.tags})
+
+            except Exception as e:
+                self.log.error(e.message + ' on keyvault: ' + resources[index]["id"])
+                pass
+
+        return r
+
+
+@KeyVaultKey.filter_registry.register('age')
+class KeyAgeFilter(AgeFilter):
+    date_attribute = "created"
```

Comment:
I though scenario is to find non-rotated keys.. In that case I assume you're interested in `updated` field or am I wrong?

Reply:
eek, specific agefilters should be scrapped in general for value filters which support it generically with value_type: age on any attribute.

Reply:
Removed, the value filter was working fine but I wasn't sure I should do a specific age filter or not (since I saw legacy age filters elsewhere) so just included it. Removing per your request. It will not be included in the next commit.